src/pages/embed/index.tsx

```diff
@@ -0,0 +1,43 @@
+import { GetServerSideProps } from 'next';
+import App from '../../App';
+import { AppHead } from '../../AppHead';
+
+/**
+ * This is a shim for now. Our app currently doesn't use
+ * next routing - we're just loading the entire app in _app.tsx
+ *
+ * This will be fixed in a later PR, adding routes with server
+ * rendering to grab OG images.
+ */
+const InspectedEmbed = () => {
+  return (
+    <>
+      <AppHead
+        title="XState Visualizer"
+        ogTitle="XState Visualizer"
+        description="Visualizer for XState state machines and statecharts"
+        importElk
+        importPrettier
+        // TODO - get an OG image for the home page
+        ogImageUrl={undefined}
+      />
+      <App />
+    </>
+  );
+};
+
+export const getServerSideProps: GetServerSideProps<{ query: any }> = async (
+  ctx,
+) => {
+  if (ctx.query.inspect === undefined) {
+    return {
+      notFound: true,
+      props: {},
+    };
+  }
+  return {
+    props: { query: ctx.query, isEmbedded: true },
+  };
+};
```

Comment:
I think we should remove this and handle this client-side if we can. Every `getServerSideProps` is a lambda call, and we should cut down on that as much as possible. All the logic in here can be handled with `useRouter`

Reply:
We can't do it on the client-side. `useRouter` first only receives the query params when it's late (after the 1st paint) so you'll see a flash of jumping layout when done in the client side.

Reply:
> We can't do it on the client-side. `useRouter` first only receives the query params when it's late (after the 1st paint) so you'll see a flash of jumping layout when done in the client side.

Can we make a wrapper for it to check the query params first? We can probably manage the flash of jumping layout by returning null for the first render inside `App.tsx`

Reply:
We can do that I suppose. we probably still need to pass `isEmbedded` as a prop to `App` to avoid parsing the url.